networkit/cpp/io/NetworkitBinaryWriter.cpp

```diff
@@ -249,74 +282,97 @@ void NetworkitBinaryWriter::write(const Graph &G, const std::string &path) {
         header.offsetWeightLists = 0;
         header.offsetWeightTranspose = 0;
     }
+
+    if(preserveEdgeIndex && weightFormat != nkbg::WEIGHT_FORMAT::NONE) {
+        header.offsetAdjIdLists = header.offsetWeightTranspose
+            + (chunks - 1) * sizeof(uint64_t)//transpWeightOffsets
+            + transpWghtOffsets.back();
+        header.offsetAdjIdTranspose = header.offsetAdjIdLists
+            + (chunks - 1) * sizeof(uint64_t)
+            + adjIndexOffsets.back();
+    } else if(preserveEdgeIndex) { //Even if there is no weight data, offset for weights are still written
```

Comment:
`//Even if there is no weight data, offset for weights are still written` -> Why is this done?